apps/mobile/src/libs/components/progress-bar/progress-bar.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { Dimensions } from "react-native";
+import { default as RNProgressBar } from "react-native-progress-step-bar";
+
+import { LinearGradient, View } from "~/libs/components/components";
+import { GradientColor } from "~/libs/enums/enums";
+import { useMemo } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+
+import { SCREEN_WIDTH_MULTIPLIER } from "./constants/constants";
+import { styles } from "./styles";
+
+type Properties = {
+	currentItemIndex: number;
+	totalItemsAmount: number;
+};
+
+const SCREEN_WIDTH: number = Dimensions.get("window").width;
+const PROGRESS_BAR_WIDTH = SCREEN_WIDTH * SCREEN_WIDTH_MULTIPLIER;
+const ZERO = 0;
+const ONE = 1;
+const DOT_DIAMETER = 12;
+const BAR_HEIGHT = 2;
+const STEP_ANIMATION_DURATION = 1000;
+
+const gradientStartCoordinates = { x: ZERO, y: ZERO };
+const gradientEndCoordinates = { x: ONE, y: ZERO };
+
+const ProgressBar: React.FC<Properties> = ({
+	currentItemIndex,
+	totalItemsAmount,
+}) => {
+	const FilledBar = useMemo(
+		() => (
+			<LinearGradient
+				colors={[...GradientColor.BLUE]}
+				end={gradientEndCoordinates}
+				start={gradientStartCoordinates}
+				style={styles.bar}
+			/>
+		),
+		[],
+	);
+
+	const Dot = useMemo(
+		() => (
+			<View style={styles.dotContainer}>
+				<LinearGradient
+					colors={[...GradientColor.BLUE]}
```

Comment:
```suggestion
					colors={GradientColor.BLUE}
```

Reply:
Can't use as `colors={GradientColor.BLUE} ` because expect `colors: (string | number)[]`

Reply:
But `GradientColor.BLUE` is `string[]`, isn't it?

Reply:
Yep, but if I use as `colors={GradientColor.BLUE}` 

received error 

`readonly ["rgba(0, 240, 255, 1)", "rgba(0, 133, 255, 1)"]
is readonly and cannot be assigned to the mutable type (string | number)[]` - reason of ` as const` in gradient-color.enum.ts. 

That's why `colors={[...GradientColor.BLUE]}`